### ISSUE TITLE
Indentation - nested case statements do not indent properly

### ISSUE THREAD

**Issue by jbodah** [elixir-lang/vim-elixir#192]
Using https://github.com/elixir-lang/vim-elixir/commit/74b5d9e74a3a1c5d61afd46183620a705b295834

Input

``` ex
defp _fetch(result, key, deep_key) do
  case _fetch(result, key) do
    {:ok, val} ->
      case _fetch(val, deep_key) do
        :error -> {:error, :deep}
        res -> res
      end

    :error -> {:error, :shallow}
  end
end
```

Visual select the block and `=` to autoindent

Output

``` ex
defp _fetch(result, key, deep_key) do
  case _fetch(result, key) do
    {:ok, val} ->
      case _fetch(val, deep_key) do
      # Should be nested one level deeper
      :error -> {:error, :deep}
      res -> res
      end

      # Should be nested one level shallower
      :error -> {:error, :shallow}
  end
end
```

Maybe hitting these is a sign I should refactor things (or at least study some vimscript to help out) 😆 

Thanks!

**Comment by kassio** [comment]
Hi there, I added some pending tests about this problem, and I'll fix it as soon as I can. 😄

**Comment by jbodah** [comment]
Fixed by #289
